### PR TITLE
[GNA] Fixes in checks, asserts, etc.

### PR DESCRIPTION
**Pull request by dorloff**

### CODE REVIEW

File: inference-engine/src/gna_plugin/gna_graph_compiler.cpp [hunk 1]
```diff
@@ -1417,6 +1417,7 @@ void GNAGraphCompiler::PermutePrimitive(InferenceEngine::CNNLayerPtr layer) {
     }
     auto layerOrder = layer->GetParamAsInts("order");
     auto quantized = InferenceEngine::getInjectedData<QuantizedLayerParams>(layer);
+    IE_ASSERT(!layer->insData.empty());
```
Comment (esmirno) [inline inference-engine/src/gna_plugin/gna_graph_compiler.cpp:1420]: i would use throw_gna_layer_exception, macro in compiler code, so it is easier to track issues related to certain layer

File: inference-engine/src/gna_plugin/gna_plugin_config.cpp [hunk 2]
```diff
@@ -71,7 +71,7 @@ void Config::UpdateFromMap(const std::map<std::string, std::string>& config) {
                 key.erase(0, 1);
                 try {
                     input_index = std::stoi(key);
-                    if (input_index < 0 | input_index > 99) {
+                    if (input_index > 99) {
```
Comment (esmirno) [inline inference-engine/src/gna_plugin/gna_plugin_config.cpp:74]: might be add informative error?

Comment (dorloff) [reply inference-engine/src/gna_plugin/gna_plugin_config.cpp:74]: We have it! See the catch section. I only removed a redundant check (an unsigned int cannot be less than 0).

File: inference-engine/src/gna_plugin/gna_graph_compiler.cpp (hunk 1 above)
Comment (dorloff) [reply inference-engine/src/gna_plugin/gna_graph_compiler.cpp:1420]: I did that in the version of PR (#903) targeted to master branch. However, we have many pieces of code like this. I propose to create a new assert-like macro for such empty pointers corresponding to layers.

Comment (esmirno) [reply inference-engine/src/gna_plugin/gna_graph_compiler.cpp:1420]: Ok for now we could use just something like IE_ASSERT(expr) << LAYER_NAME(layer),  or create GNA_LAYER_ASSERT(expr, layer) which effectively transformed to previous code

Comment (dorloff) [reply inference-engine/src/gna_plugin/gna_graph_compiler.cpp:1420]: Yes, but let's leave it this way in the release branch if you are not against that.

**Pull request merged**